src/cls/tabular/fbreader.cc

```diff
@@ -0,0 +1,256 @@
+  if( vm.count( "help" ) ) {
+    std::cout << all_opts << std::endl ;
+    return 1;
+  }
+  po::notify( vm ) ;
+
+  do_read( debug, oid, pool ) ;
+
+  return 0 ;
+}
+
+// ========== //
+//   DO READ  //
+// ========== //
+void do_read( bool debug, 
+              std::string oid,
+              std::string pool ) {
+
+  if( debug ) {
+    std::cout << "in do_read..." << std::endl ;
+    std::cout << "oid         : " << oid         << std::endl ;
+    std::cout << "pool        : " << pool        << std::endl ;
+  }
+
+  // connect to rados
+  librados::Rados cluster;
+  cluster.init(NULL);
+  cluster.conf_read_file(NULL);
+  int ret = cluster.connect();
+
+  // open pool
+  librados::IoCtx ioctx ;
+  ret = cluster.ioctx_create( pool.c_str(), ioctx ) ;
+
+  // read bl_seq
+  librados::bufferlist wrapped_bl_seq ;
+  int num_bytes_read = ioctx.read( oid.c_str(), wrapped_bl_seq, (size_t)0, (uint64_t)0 ) ;
+
+  if( debug ) {
+    std::cout << "num_bytes_read : " << num_bytes_read << std::endl ; 
+  }
+
+  ceph::bufferlist::iterator it_wrapped = wrapped_bl_seq.begin() ;
+
+  // ================================================================================ //
+  // display data
+
+  while( it_wrapped.get_remaining() > 0 ) {
+
+    if( debug ) {
+      std::cout << "it_wrapped.get_remaining() = " << it_wrapped.get_remaining() << std::endl ;
+    }
+
+    // grab the Root
+    ceph::bufferlist bl ;
+    ::decode( bl, it_wrapped ) ; // this decrements get_remaining by moving iterator
+    const char* fb = bl.c_str() ;
+
+    auto root      = Tables::GetRoot( fb ) ;
+    auto data_type = root->relationData_type() ;
+
+    if( debug ) {
+      std::cout << "data_type : " << data_type << std::endl ;
+    }
+
+    // process one Root>Rows flatbuffer
+    if( data_type == Tables::Relation_Rows ) {
+      if( debug ) {
+        std::cout << "if data_type == Tables::Relation_Rows" << std::endl ;
+      }
+
+      //auto schema    = root->schema() ;
+      //if( debug ) {
+      //  std::cout << "schema : " << std::endl ;
+      //  for( unsigned int i = 0; i < schema->Length(); i++ )
+      //    std::cout << (unsigned)schema->Get(i) << std::endl ;
+      //}
+ 
+      auto rows = static_cast< const Tables::Rows* >( root->relationData() ) ;
+      auto table_name_read = rows->table_name() ;
+
+      //auto schema_read     = rows->schema() ;
+      auto nrows_read      = rows->nrows() ;
+      auto ncols_read      = rows->ncols() ;
+      auto rids_read       = rows->RIDs() ;
+      auto rows_data       = rows->data() ; // [ Record ]
+
+      if( debug ) {
+        std::cout << "table_name_read->str() : " << table_name_read->str() << std::endl ;
+        //std::cout << "schema_read->Length() : " << schema_read->Length() << std::endl ;
+        std::cout << "nrows_read     : " << nrows_read     << std::endl ;
+        std::cout << "ncols_read     : " << ncols_read     << std::endl ;
+      }
+
+      // print schema to stdout
```

Comment:
do you want to uncomment for debug flag?

Reply:
These lines actually break when testing the Col schema instead of Rows. Need to remove/rework these.